crates/ruff_server/src/lint.rs

```diff
@@ -274,14 +264,12 @@ fn to_lsp_diagnostic(
                 title: suggestion.unwrap_or(name).to_string(),
                 noqa_edit,
                 edits,
-                code: rule.noqa_code().to_string(),
+                code: code?.to_string(),
```

Comment:
nit: we could short-circuit if `code` is `None` and avoid generating the edits although I think currently it's only the syntax errors that doesn't have code and those doesn't have fixes, so maybe this isn't that useful